tests/test_constrained.py

```diff
@@ -949,3 +949,26 @@ def test_variable_multiple_times(self):
                 self.assertAlmostEqual(cmodel.lhs.get_linear('x0'), 3.5,
                                        msg='constraint c1, linear(x0) should be 3.5')
 
+    def test_functional(self):
```

Comment:
Need tests for:
- different variable labels sets, e.g. strings, tuples, frozensets
- empty CQM
- no constraints

Reply:
Also needs a test for the `a + b + 3 <= 0` and `a + b <= -3` cases.

Reply:
ok, tests for empty CQM, no constraints, only quadratic objective and constraint are in. The problem with tuples and forzensets as variables is that their representation contains characters that are not allowed by the LP format. In that case an error is raised, pointing to the variable that has bad characters.